app/Services/AppointmentService.php

```diff
@@ -46,4 +48,31 @@ public function checkAppointment($approvedAppointments)
             abort(Response::HTTP_BAD_REQUEST, "there is an already approved appointment at this time");
         }
     }
+
+    public function reject($data)
+    {
+        $data['status'] = AppointmentStatus::REJECTED;
+        $appointment = $this->repository->find(request()->appointment);
```

Comment:
Get the appointment from the controller and pass it to the service

Reply:
done